common/models/report-model.js

```diff
@@ -1286,7 +1286,178 @@ module.exports = function (ReportModel) {
+                        options,
+                    });
+                    return Promise.reject('Product already present in stock order');
+                } else {
+                    return Promise.resolve();
+                }
+        }).catch(function (error) {
+            logger.error({
+                functionName: 'addProductToStockOrder',
+                message: 'Error while searching existing product in stock order',
+                options,
+                error
+            });
+            return Promise.reject(error);
+        })
+            .then(function () {
+                logger.debug({
+                    functionName: 'addProductToStockOrder',
+                    message: 'Will find inventory associated with product, store and org',
+                    options,
+                });
+          return ReportModel.app.models.InventoryModel.find({
+              where: {
+                  productModelId: product.id,
+                  orgModelId: id,
+                  storeModelId
+              }
+          });
+        }).then(function (inventoryInstances) {
+                var row;
+
+                    var useRow = true;
+                    var caseQuantity, quantityOnHand, desiredStockLevel, orderQuantity;
+                    if (product.tags) {
+                        var tagsAsCsv = product.tags.trim();
+                        //logger.debug({ tagsAsCsv: tagsAsCsv });
+                        var tagsArray = tagsAsCsv.split(',');
+                        if (tagsArray && tagsArray.length>0) {
+                            _.each(tagsArray, function (tag) {
+                                tag = tag.trim();
+                                if (tag.length>0) {
+                                    //logger.debug({ tag: tag });
+                                    // http://stackoverflow.com/questions/8993773/javascript-indexof-case-insensitive
+                                    var prefix = 'CaseQuantity:'.toLowerCase();
+                                    if (tag.toLowerCase().indexOf(prefix) === 0) {
+                                        var caseQty = tag.substr(prefix.length);
+                                        //logger.debug({ message: `based on a prefix, adding CaseQuantity: ${caseQty}` });
+                                        caseQuantity = Number(caseQty);
+                                    }
+                                    else {
+                                        //logger.debug({ message: 'ignoring anything without a prefix' });
+                                    }
+                                }
+                            });
+                        }
+                    }
+                    var inventory = _.find(inventoryInstances, function (eachInventory) {
+                        return eachInventory.productModelId.toString() === product.id.toString();
+                    });
+                    if (!inventory) {
```

Comment:
Don't need this check here. User can add product if there's no inventory in store